nebula-logger/main/logger-engine/classes/LogEntryEventBuilder.cls

```diff
@@ -408,21 +408,28 @@ global with sharing class LogEntryEventBuilder {
      * @return        The same instance of `LogEntryEventBuilder`, useful for chaining methods
      */
     global LogEntryEventBuilder setRecord(SObject record) {
-        if (this.shouldSave == false || record == null) {
+        if (this.shouldSave == false) {
             return this;
         }
 
-        String recordJson = getJson(record);
-        String cleanedRecordJson = applyDataMaskRules(recordJson, false, true);
-        Boolean recordJsonMasked = cleanedRecordJson != recordJson;
-
         this.logEntryEvent.RecordCollectionType__c = 'Single';
-        this.logEntryEvent.RecordId__c = record.Id;
-        this.logEntryEvent.RecordJson__c = cleanedRecordJson;
-        this.logEntryEvent.RecordJsonMasked__c = recordJsonMasked;
-        this.logEntryEvent.RecordSObjectClassification__c = getSObjectClassification(record.getSObjectType());
-        this.logEntryEvent.RecordSObjectType__c = record.getSObjectType().getDescribe().getName();
-        this.logEntryEvent.RecordSObjectTypeNamespace__c = getSObjectTypeNamespace(record.getSObjectType());
+
+        if (record == null) {
+            this.logEntryEvent.RecordJson__c = getJson(record);
```

Comment:
since both branches are calling `getJson(record)` should that be consolidated outside the if/else ?

Reply:
Yup, good call, I'll put it into a `String` variable just before the `if/else` statement